listenbrainz/db/missing_musicbrainz_data.py

```diff
@@ -0,0 +1,115 @@
+# listenbrainz-server - Server for the ListenBrainz project.
+#
+# Copyright (C) 2020 Vansika Pareek <vansikapareek2001@gmail.com>
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+import ujson
+import sqlalchemy
+
+from listenbrainz import db
+from flask import current_app
+
+
+def insert_user_missing_musicbrainz_data(user_id, data, source):
+    """ Insert missing musicbrainz data that a user has submitted to ListenBrainz but
+        has not submitted to MusicBrainz in the db.
+
+        Args:
+            user_id (int): row id of the user.
+            data (list): Data that is submitted to ListenBrainz by the users
+                         but is not submitted to MusicBrainz.
+            source (str): Source of generation of missing MusicBrainz data.
+    """
+    missing_musicbrainz_data = {
+        'missing_musicbrainz_data': data
+    }
+
+    with db.engine.connect() as connection:
+        connection.execute(sqlalchemy.text("""
+            INSERT INTO missing_musicbrainz_data (user_id, data, source)
+                 VALUES (:user_id, :missing_musicbrainz_data, :source)
+            ON CONFLICT (user_id)
+          DO UPDATE SET user_id = :user_id,
+                        data = :missing_musicbrainz_data,
+                        source = :source,
+                        created = NOW()
+            """), {
+                'user_id': user_id,
+                'missing_musicbrainz_data': ujson.dumps(missing_musicbrainz_data),
```

Comment:
I would prefer to have a pydantic model for this data field. Otherwise, it's a pain to understand what data is in the column.